scripts/cp-pkgs.js

```diff
@@ -68,18 +88,96 @@ function getAssetEntry(asset) {
   }
   const destDir = path.join(assetPkg, 'dist', path.basename(asset));
-  return cpFile(asset, destDir).then(() => console.log(`cp ${asset} -> ${destDir}`));
+  return cpFile(asset, destDir).then(
+    () => console.log(`cp ${asset} -> ${destDir}`),
+    (err) => {
+      throw err;
+    }
+  );
 }
 
-cleanPkgDistDirs();
+/**
+ * Imports all files in index.d.ts and compiles a bundled .d.ts file for
+ * UMD files.
+ */
+function dtsBundler() {
+  const packageDirectories = fs.readdirSync(D_TS_DIRECTORY);
+  packageDirectories.forEach((packageDirectory) => {
+    const main = path.resolve(D_TS_DIRECTORY, packageDirectory, './index.d.ts');
+    dts.bundle({
+      name: packageDirectory,
+      main,
+    });
+  });
+}
 
-Promise.all(globSync('build/*.{css,js,map}').map(cpAsset)).catch((err) => {
-  console.error(`Error encountered copying assets: ${err}`);
-  process.exit(1);
-});
+/**
+ * @param {string} asset filepath relative to the root directory
+ * @param {string} assetPkg directory path to destination package
+ * @returns {string} destination filepath for UMD declaration file.
+ */
+function getDeclarationFileName(asset, assetPkg) {
+  const packageName = path.parse(asset).name.replace(/^mdc-|\.d$/g, '');
+  const isAllInOne = packageName === ALL_IN_ONE_PACKAGE;
+  const destFileName = isAllInOne ? packageName : `mdc.${toCamelCase(packageName)}`;
+  return path.join(assetPkg, 'dist', `${destFileName}.d.ts`);
+}
+
+/**
+ * Copies the declaration file from the /build directory to its respective
+ * destination directory.
+ * @param {string} asset filepath relative to the root directory
+ * @returns {Promise<void>}
+ */
+async function cpDeclarationAsset(asset) {
+  const assetPkg = path.parse(asset.split('build/')[1]).dir;
+  if (!fs.existsSync(assetPkg)) {
+    return Promise.reject(new Error(`Non-existent asset package path ${assetPkg} for ${asset}`));
+  }
+
+  const destDir = getDeclarationFileName(asset, assetPkg);
+  return cpFile(asset, destDir).then(
+    () => console.log(`cp ${asset} -> ${destDir}`),
+    (err) => {
+      throw err;
+    }
+  );
+}
+
+/**
+ * 1. Cleans all /dist directories in each package
+ * 2. Copies generated css, js, and map files to the respective packages
+ * 3. Bundles the declaration files into one file for the UMD index.js file
+ * 4. Copies the generated declaration file from step 3, into the respective pacakge
+ */
+function main() {
+  cleanPkgDistDirs();
+
+  Promise.all(globSync('build/*.{css,js,map}').map(cpAsset)).catch((err) => {
+    console.error('Error encountered copying assets:', err);
+    process.exit(1);
+  });
+
+  // this method builds the files that the next lines copy to each package
+  dtsBundler();
+
+  Promise.all(
+    globSync(`build/packages/**/{${DECLARATION_FILE_PREFIX},${ALL_IN_ONE_PACKAGE}}*.d.ts`)
+      .map(cpDeclarationAsset)
+  ).catch((err) => {
+    console.error('Error encountered copying assets:', err);
+    process.exit(1);
+  });
+}
+
+main();
```

Comment:
Nit: Is there even a reason for `main` to be a function encapsulating this code if we're just immediately executing it anyway?

Reply:
I thought it cleaned it up by putting it in a function. I can move it out.